### ISSUE TITLE
[Remote-SSH Bug]: Could not fetch remote environment

### ISSUE THREAD

**Issue by ArcticOc** [microsoft/vscode-remote-release#10351]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [x] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It doesn’t connect successfully (same error message)

### Steps To Reproduce

1. connect to a remote server
2. open a folder
3. error message

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
ExtensionInstallFailed(ExtensionInstallFailed("file:///home/trist/.vscode-server/cli/servers/Stable-d78a74bcdfad14d5d3b1b782f87255d802b57511/server/out/vs/server/node/server.main.js:33\n\t| KEY ('==' | '!=' | '<' | '<=' | '>' | '>=' | 'in' | 'not' 'in') value`,this.D())}}u(){const t=this.D();switch(t.type){case 17:case 18:return this.z(),t.lexeme;case 11:return this.z(),\"true\";case 12:return this.z(),\"false\";case 13:return this.z(),\"in\";default:return\"\"}}w(t){return t.replaceAll(this.v,\"\")}x(){return this.f[this.g-1]}y(t){return this.C(t)?(this.z(),!0):!1}z(){return this.E()||this.g++,this.x()}A(t,s){if(this.C(t))return this.z();
```

</p>
</details>

### Anything else?

After updating to version 1.94, this issue occurred.

**Comment by joshspicer** [comment]
Could you please provide the Remote SSH log as outlined [here](https://github.com/microsoft/vscode-remote-release/wiki/Remote-SSH-troubleshooting)

**Comment by ArcticOc** [comment]
> Could you please provide the Remote SSH log as outlined [here](https://github.com/microsoft/vscode-remote-release/wiki/Remote-SSH-troubleshooting)

Here is the log of server.

<details>
<summary>Remote-SSH Log</summary>

```
*
* Visual Studio Code Server
*
* By using the software, you agree to
* the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
* the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
*
Server bound to /tmp/code-86ce8cc3-e30d-4de2-96ad-cf936116256b
Extension host agent listening on /tmp/code-86ce8cc3-e30d-4de2-96ad-cf936116256b

[03:54:31] 




[03:54:31] Installing extensions...
[03:54:31] Extension host agent started.
[03:54:31] [<unknown>][91ae8969][ExtensionHostConnection] New connection established.
[03:54:31] [<unknown>][cd5591cf][ManagementConnection] New connection established.
[03:54:31] Extension 'christian-kohler.path-intellisense' v2.9.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'christian-kohler.path-intellisense@1.2.3'.
[03:54:31] Extension 'alefragnani.bookmarks' v13.5.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'alefragnani.bookmarks@1.2.3'.
[03:54:31] Extension 'valentjn.vscode-ltex' v13.1.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'valentjn.vscode-ltex@1.2.3'.
[03:54:31] Extension 'ms-vscode.hexeditor' v1.10.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-vscode.hexeditor@1.2.3'.
[03:54:31] Extension 'psioniq.psi-header' v1.23.1 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'psioniq.psi-header@1.2.3'.
[03:54:31] Extension 'mhutchie.git-graph' v1.30.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'mhutchie.git-graph@1.2.3'.
[03:54:31] Extension 'vscodevim.vim' v1.28.1 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'vscodevim.vim@1.2.3'.
[03:54:31] Extension 'pkief.material-product-icons' v1.7.1 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'pkief.material-product-icons@1.2.3'.
[03:54:31] Extension 'jock.svg' v1.5.4 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'jock.svg@1.2.3'.
[03:54:31] Extension 'chouzz.vscode-better-align' v1.4.2 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'chouzz.vscode-better-align@1.2.3'.
[03:54:31] Extension 'ms-vscode.remote-explorer' v0.4.3 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-vscode.remote-explorer@1.2.3'.
[03:54:31] Extension 'bierner.markdown-preview-github-styles' v2.1.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'bierner.markdown-preview-github-styles@1.2.3'.
[03:54:31] Extension 'ms-toolsai.jupyter-renderers' v1.0.19 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-toolsai.jupyter-renderers@1.2.3'.
[03:54:31] Extension 'ms-toolsai.vscode-jupyter-cell-tags' v0.1.9 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-toolsai.vscode-jupyter-cell-tags@1.2.3'.
[03:54:31] Extension 'ms-toolsai.vscode-jupyter-slideshow' v0.1.6 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-toolsai.vscode-jupyter-slideshow@1.2.3'.
[03:54:31] Extension 'ms-toolsai.vscode-jupyter-powertoys' v0.1.1 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-toolsai.vscode-jupyter-powertoys@1.2.3'.
[03:54:31] Extension 'ms-vscode-remote.remote-ssh-edit' v0.87.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-vscode-remote.remote-ssh-edit@1.2.3'.
[03:54:31] Extension 'ms-vscode-remote.remote-ssh' v0.115.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-vscode-remote.remote-ssh@1.2.3'.
[03:54:31] Extension 'eamodio.gitlens' v15.5.1 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'eamodio.gitlens@1.2.3'.
[03:54:31] Extension 'be5invis.vscode-custom-css' v7.3.1 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'be5invis.vscode-custom-css@1.2.3'.
[03:54:31] Extension 'james-yu.latex-workshop' v10.4.2 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'james-yu.latex-workshop@1.2.3'.
[03:54:31] Extension 'github.remotehub' v0.64.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'github.remotehub@1.2.3'.
[03:54:31] Extension 'ms-vscode.remote-repositories' v0.42.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-vscode.remote-repositories@1.2.3'.
[03:54:31] Extension 'davidanson.vscode-markdownlint' v0.56.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'davidanson.vscode-markdownlint@1.2.3'.
[03:54:31] Extension 'ms-toolsai.jupyter-keymap' v1.1.2 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-toolsai.jupyter-keymap@1.2.3'.
[03:54:31] Extension 'ms-vscode-remote.remote-wsl' v0.88.4 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-vscode-remote.remote-wsl@1.2.3'.
[03:54:31] Extension 'ms-toolsai.jupyter' v2024.10.2024100401 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-toolsai.jupyter@1.2.3'.
[03:54:31] Extension 'ms-python.python' v2024.16.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-python.python@1.2.3'.
[03:54:31] Extension 'ms-python.vscode-pylance' v2024.9.2 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'ms-python.vscode-pylance@1.2.3'.
[03:54:31] Extension 'github.copilot' v1.235.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'github.copilot@1.2.3'.
[03:54:31] Extension 'charliermarsh.ruff' v2024.50.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'charliermarsh.ruff@1.2.3'.
[03:54:31] Extension 'github.copilot-chat' v0.21.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'github.copilot-chat@1.2.3'.
[03:54:31] Extension 'mechatroner.rainbow-csv' v3.12.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'mechatroner.rainbow-csv@1.2.3'.
[03:54:31] Extension 'orangex4.hsnips' v0.4.8 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'orangex4.hsnips@1.2.3'.
[03:54:31] Extension 'usernamehw.errorlens' v3.20.0 is already installed. Use '--force' option to update to latest version or provide '@<version>' to install a specific version, for example: 'usernamehw.errorlens@1.2.3'.
[03:54:32] [<unknown>][91ae8969][ExtensionHostConnection] <3098539> Launched Extension Host Process.
[03:54:40] [<unknown>][cd5591cf][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[03:54:46] [<unknown>][91ae8969][ExtensionHostConnection] <3098539> Extension Host Process exited with code: 0, signal: null.
Cancelling previous shutdown timeout
[03:54:46] Cancelling previous shutdown timeout
Last EH closed, waiting before shutting down
[03:54:46] Last EH closed, waiting before shutting down
```

**Comment by dx2ztm76-new** [comment]
I encountered the same issue. I tried removing .vscode-server on the remote host and .vscode on my MacBook, then reinstalling version 1.94, but it didn't work. 
The problem disappeared after downgrading to version 1.93.1.

**Comment by GPhilo** [comment]
I'm also observing the same issue. I tried downgrading the Remote - SSH extension version to v0.144.3, but the issue persists.

A (very poor) workaround I found, is to kill the running server after identifying its PID from the Remote - SSH output log, e.g.:
```
[16:42:10.117] [server] Found running server (pid=26510)
[16:42:10.117] [server] Found running server (pid=26510)
[16:42:10.117] [server] Found running server (pid=26510)
[16:42:10.117] [server] Found running server (pid=26510)
[16:42:10.117] [server] Found running server (pid=26510)
[16:42:10.118] [server] Installing extensions...
[16:42:10.118] [server] Installing extensions...
[16:42:10.118] [server] Installing extensions...
[16:42:10.118] [server] Installing extensions...
[16:42:10.118] [server] Installing extensions...
```
Run `kill 26510` after ssh-ing into your remote machine, then reload the window and it should load normally.
Unfortunately, at the next reload the issue will reappear and a manual kill is again needed.

**Comment by mthorpe7** [comment]
#10376 has a workaround in - set `"remote.SSH.defaultExtensions": []` (or comment out your existing extensions).

**Comment by ArcticOc** [comment]
This issue has been fixed in version 1.94.2.
